main.go

```diff
@@ -246,3 +284,77 @@ func main() {
 		os.Exit(1)
 	}
 }
+
+// GetTLSOptionOverrideFuncs returns a list of TLS configuration overrides to be used
+// by the webhook server.
+func GetTLSOptionOverrideFuncs(options TLSOptions) ([]func(*tls.Config), error) {
+	var tlsOptions []func(config *tls.Config)
+	var TLSMinVersionIndex, TLSMaxVersionIndex int
+	for tlsVersionIndex, TLSVersion := range cliflag.TLSPossibleVersions() {
+		if options.TLSMinVersion == TLSVersion {
+			TLSMinVersionIndex = tlsVersionIndex
+		}
+		if options.TLSMaxVersion == TLSVersion {
+			TLSMaxVersionIndex = tlsVersionIndex
+		}
+	}
+	// Check if tls min version flag value is greater than max version
+	if TLSMinVersionIndex > TLSMaxVersionIndex &&
+		TLSMaxVersionIndex != -1 &&
+		TLSMinVersionIndex != -1 {
+		return nil, fmt.Errorf("TLS version flag min version is greater than max version")
+	}
+
+	tlsMinVersion, err := cliflag.TLSVersion(options.TLSMinVersion)
+	if err != nil {
+		return nil, err
+	}
+	tlsOptions = append(tlsOptions, func(cfg *tls.Config) {
+		cfg.MinVersion = tlsMinVersion
+	})
+
+	tlsMaxVersion, err := cliflag.TLSVersion(options.TLSMaxVersion)
+	if err != nil {
+		return nil, err
+	}
+	tlsOptions = append(tlsOptions, func(cfg *tls.Config) {
+		cfg.MaxVersion = tlsMaxVersion
+	})
+	// Cipher suites should not be set if empty.
+	if options.TLSMinVersion == "VersionTLS13" && options.TLSMaxVersion == "VersionTLS13" {
+		setupLog.Info("warning: cipher suite flag set to null for TLS version 1.3")
+		options.TLSCipherSuites = ""
+	}
```

Comment:
Is this correct? Should the comment be that "Cipher suites should not be set for TLS13"? Why log a warning when doing this? It would make more sense to only log the warning if the user tried setting cipher suites AND required TLS13.